http-server/src/test/java/org/triplea/modules/game/ConnectivityControllerTest.java

```diff
@@ -32,7 +34,9 @@ void checkConnectivityNegativeCase() {
   @Test
   void checkConnectivityPositiveCase() throws IOException {
     openSocket();
-    final boolean result = verifyEndpointReturningObject(client -> client.checkConnectivity(PORT));
+    // TODO Setup GameEntry so that the server correctly checks the connectivity.
```

Comment:
The game entry and connectivity check happen effectively in parallel. There is no guarantee that the game entry will have been created. I think the IP to check needs to become an optional parameter to the connectivity check.

Reply:
Well the main difference now is that we no longer pass an IP/port to this endpoint.
We just pass a GameID, which then checks the LobbyGame entry for this id and checks the endpoint passed there.

Maybe merging this endpoint with the game entry endpoint?

Reply:
Hmm, I'd be concerned about a separation of concerns. It would be nice though to get that fast feedback when a game is posted that it cannot be accessed.

What do you think about creating a second connectivity check endpoint and invoke that one with a slightly different payload?

I didn't look at the full diff, @RoiEXLab , we're passing game id now? That might not be wise as there is no guarantee that create game *will* be invoked first, they happen on different threads.

Reply:
@RoiEXLab having create game detect connectivity is pretty interesting. The error handling would be a bit more complex, as we can have multiple scenarios why a game could not be created. It certainly has the benefit though that a user does not wait the 5 or so seconds for the connectivity check to fire. It's worth investigating, and could lead to the removal of the connectivity check endpoint, which would be simpler.

Reply:
@RoiEXLab thinking about it a bit more, another consideration - gameId is assigned by the serve when a game is created. You can't / don't pass a game id when creating a game. The server can also re-assign gameIds as well, or it is allowed to lose gameIds in which case a keep-alive will see a new gameId has been assigned.

FWIW, passing in IP address to a new connectivity check endpoint, or doing the check on game create seem like the two best options you may have.

Reply:
Ok @DanVanAtta coming back at this PR I think I'll go with the merge post and verify step option. However I have one question:
How do I change `LobbyWatcherFeignClient#postGame(@HeaderMap Map<String, Object> headers, LobbyGame lobbyGame)` to return a more complex object than string to signalize that something went wrong with the verification?
Or should I just use a different HTTP status code for this?

Reply:
Simply convert the return type to an object that can be converted to a JSON.

Could you walk me through briefly your reasons for re-using `postGame`?

It is a trade-off, FWIW this is the pro/con list I can think of:
Pro:
  - Can remove background thread that does connectivity check
  - Connectivity failures could become more responsive

Con:
  - There are at least two places where we post games. Both would need to handle the updated payload
  - The error code handling gets a bit tricky. First you need to be sure that the game id is returned null and you return an error code that indicates to the client the situation. This level of error checking / API contract is not the worst but it is intrinsically more complex
  - have to be careful of timing, the new game host server port may not yet be available until well after the 'post' call has completed. We'll need to find out if that was the reason for the connectivity check to have had the background thread and the very long delay before it is executed.
  - Break compatibility
  - Could maybe lead us down a pattern of having postGame return an even more complex return payload. Can be a slippery slope to violating  [interface segragation principle](https://en.wikipedia.org/wiki/Interface_segregation_principle).
  - Server side, every time a game is posted it'll always do a connectivity check, even if we do not want it. Clients reconnecting for example should not need the connectivity check. A callback from a 'postGame' endpoint could be considered an odd side effect.

Thinking about the timing issue now, and the last point of 'side-effects', and ease of update, I'd personally augment the connectivity check endpoint. I'm curious if you've different thoughts on this, if you see the pro list as much larger than I listed and perhaps the cons easier to solve than stated.